src/paradigma/heart_rate/heart_rate_analysis.py

```diff
@@ -64,25 +65,84 @@ def signal_quality_classification(df: pd.DataFrame, config: SignalQualityClassif
 
     # Prepare the data
     lr_clf.feature_names_in_ = ['var', 'mean', 'median', 'kurtosis', 'skewness', 'f_dom', 'rel_power', 'spectral_entropy', 'signal_to_noise', 'auto_corr']
-    X = df.loc[:, lr_clf.feature_names_in_]
+    X = df_windowed.loc[:, lr_clf.feature_names_in_]
 
     # Normalize features using mu and sigma
     X_normalized = X.copy()
     for idx, feature in enumerate(lr_clf.feature_names_in_):
         X_normalized[feature] = (X[feature] - mu[idx]) / sigma[idx]
```

Comment:
I think this the can be serialized using `numpy` or `scipy`, but perhaps for another point in time.

Reply:
Apologies, you made this suggestion before and I promised to fix this in this PR but totally forgot. Will update this ;)